components/gitpod-protocol/src/teams-projects-protocol.ts

```diff
@@ -21,6 +21,8 @@ export interface ProjectSettings {
     allowUsingPreviousPrebuilds?: boolean;
     // how many commits in the commit history a prebuild is good (undefined and 0 means every commit is prebuilt)
     prebuildEveryNthCommit?: number;
+    // preferred workspace classes
+    workspaceClasses?: WorkspaceClasses;
```

Comment:
This is bit of a smell. Why is it plural workspaceClasses when we're not passing in a list?

Currently we look to only have a single preference, could we go with a list here (which is initially only 1 item long) and allow for the "preferences" to be a list?

I can imagine wanting to specify a priority list of classes I want to use. This also facilitates gradual migration away from one WS class to another over time - by changing the priorities (ordering)

Reply:
I think it make senses to reuse the shape and naming we already have in place. Although I agree `WorkspaceClassConfig` or similar would have been a better choice in the first place.